### ISSUE TITLE
[A11y] Badge improvement request

### ISSUE THREAD

**Issue by mendeza** [ITS-HCD/nysds#1468]
# Overview

The Badge component plays a variable and hybrid role in a UI composition. It can be interpreted as a sort of mini header. It can serve as a sort of metadata. It can seem like a sort of mini static alert. Though the Badge is a container for a word or a phrase, it has significant communicative non-text content, namely color and iconography. In particular, the use of color is directly tied to the "intent" of a Badge, which ranges in a scale of intensity/severity from neutral to "dangerous." Optional iconography can be arbitrary or further intensify the intent. As these are only present in the visual dimension, the lack of corresponding means of expression for those with sight disabilities is a significant shortcoming.

## WCAG violations

- [SC 1.3.1 A](https://www.w3.org/WAI/WCAG22/quickref/#info-and-relationships) Info and Relationships: Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text.
- [SC 1.3.3 A](https://www.w3.org/WAI/WCAG22/quickref/#sensory-characteristics) Sensory Characteristics: Instructions provided for understanding and operating content do not rely solely on sensory characteristics of components such as shape, color, size, visual location, orientation, or sound.
- [SC 1.3.6 AAA](https://www.w3.org/WAI/WCAG22/quickref/#identify-purpose) Identify Purpose: In content implemented using markup languages, the purpose of user interface components, icons, and regions can be programmatically determined.
- [SC 1.4.1 A](https://www.w3.org/WAI/WCAG22/quickref/#use-of-color) Use of Color: Color is not used as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element.

## Requests
See: #1095 for exploration, rationale

* Refactor the rendered Badge HTML to the native `<mark>` element
  * This performs the minimum-needed distinction from surrounding text
* Add an optional prop for additional context; inject this in an `.sr-only` text element placed in the DOM after the visible Badge label text
  * This can be used by authors to arbitrarily "add color" and more fully communicate intent to screen reader users
